packages/babel-plugin-transform-modules-commonjs/src/index.ts

```diff
@@ -235,6 +235,12 @@ export default declare((api, options) => {
 
           ensureStatementsHoisted(headers);
           path.unshiftContainer("body", headers);
+          path.get("body").forEach(path => {
+            if (headers.indexOf(path.node) === -1) return;
+            if (path.isVariableDeclaration()) {
+              path.scope.registerDeclaration(path);
+            }
```

Comment:
Note for reviewers: this code is the same as https://github.com/babel/babel/blob/bee3e358515d0f3a67c557fad9a9afa0fe929a5e/packages/babel-core/src/transformation/file/file.ts#L217-L223